stacker/stack.py

```diff
@@ -72,7 +76,7 @@ def __init__(self, definition, context, variables=None, mappings=None,
         self.force = force
         self.enabled = enabled
         self.protected = protected
-        self.context = copy.deepcopy(context)
+        self.context = context
```

Comment:
I really don't know what I can do here, without this change, tests in Python 3 will not pass, complaining something cannot be pickled. I've done some research, it might have something to do with our use of `schematics`.

Can we have a good test case and fix the memory issue instead?

Reply:
#111 is the original issue, though strangely I think my fix (that is linked in that issue) was rolled back by #550.  @ejholmes do you remember why you might have rolled it back, and why it no longer causes issues with memory growth?  It might be the memory issue has been resolved entirely because of the DAG work.

Reply:
Ok, I created a quick script to test if memory usage changed after removing the deepcopy:

```python
import sys
import resource

from stacker.context import Context
from stacker.config import Config, Stack


def generate_context(stacks, conf):
    stack_names = [d["name"] for d in stacks]
    stack_objs = [Stack(d) for d in stacks]

    conf.stacks = stack_objs
    conf.validate()
    return Context(
        environment={},
        stack_names=stack_names,
        config=conf,
    )


conf = Config()
conf.namespace = "test-namespace"

stacks = []
STACK_COUNT = int(sys.argv[1])

for i in range(STACK_COUNT):
    stacks.append({"name": "vpc%s" % i, "class_path": "blueprints.VPC"})


start_mem = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss
for i in range(len(stacks)):
    ii = i + 1
    s = stacks[:ii]
    ctx = generate_context(s, conf)

stacks_created = ctx.get_stacks()

end_mem = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss
total_mem = end_mem - start_mem

print total_mem,
```

And then ran it repeatedly, increasing the config size by one stack each time with this shell script, and dumped the output into a csv format file:

```shell
#!/bin/bash

ROWS=""

for i in $(seq 100)
do
    echo "$i,$(python ./test_stacker_mem.py $i)"
done
```

That resulted in this graph:

<img width="953" alt="screenshot 2018-06-23 15 36 16" src="https://user-images.githubusercontent.com/517334/41814200-3fccb500-76fb-11e8-994d-5d43cfeb6e1b.png">

So it looks like in 1.3 we regressed and started using more memory once more, and this should help fix it.  @ejholmes unless you can think of a reason this might be bad, I'm going to go ahead and go witht his too.

(BTW, here's the google spreadsheet with my run's results: https://docs.google.com/spreadsheets/d/15rP-RPYg7DtFQIwB_lK6oaJwPQHG6tz1NfxsvUKVJeM/edit?usp=sharing)

Reply:
We already talked about this offline, but I feel like this should be fine, and removing the deepcopy is the right thing to do. May have some unintended side effects for anyone doing something crazy `self.context` in custom blueprints.